system_tests/monitoring.py

```diff
@@ -26,6 +26,7 @@
 
 retry_404 = RetryErrors(NotFound, max_tries=5)
 retry_404_500 = RetryErrors((NotFound, InternalServerError))
+retry_400_500 = RetryErrors((BadRequest, InternalServerError))
 retry_500 = RetryErrors(InternalServerError)
 retry_503 = RetryErrors(ServiceUnavailable)
```

Comment:
This collection of retries seems to be a blunt instrument.

Can we refine them to use `error_predicate` and actually sniff around for the expected state of the error? (doesn't have to be in this PR)